exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -286,6 +295,7 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 	resourceSetBinding := clusterResourceSetBinding.GetOrCreateBinding(clusterResourceSet)
 
 	// Iterate all resources and apply them to the cluster and update the resource status in the ClusterResourceSetBinding object.
+	resourceNotFound := false
```

Comment:
Q: Should we add a comment explaining why we are requeing resources not found?

Reply:
In general yes.

The question is if we still want to do it.

Initially I added this because through a bug in the new CR version I was hitting a stale cache with get CM/Secret. This doesn't happen anymore since I fixed CR.

Now the only reason to requeue is basically to get a quick requeue if someone deploys first the CRS and then the referenced CM/Secret.

WDYT should we requeue for those cases?

If we don't the next resync will catch it.

Reply:
Thougt a bit more about it. The requeue doesn't make sense (now that we are not caching CM/Secret anymore).

The controller is watching on all ConfigMaps / Secrets, so we don't need a requeue for the case a ConfigMap / Secret is created later on.